Solutions/Salesforce Service Cloud/Analytic Rules/Salesforce-SigninsMultipleCountries.yaml

```diff
@@ -0,0 +1,43 @@
+id: 3094e036-e5ae-4d6e-8626-b0f86ebc71f2
+name: Brute force attack against user credentials
+description: |
+  'This query searches for successful user logins from different countries within 3 hours.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: SalesforceServiceCloud
+    dataTypes:
+      - SalesforceServiceCloud
+queryFrequency: 20m
+queryPeriod: 20m
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1078
+query: |
+  let timeframe = ago(3h);
+  let threshold = 2;
+  let Countrydb = externaldata(Network:string, geoname_id:string, continent_code:string, continent_name:string, country_iso_code:string, country_name:string)
+  [@"https://raw.githubusercontent.com/datasets/geoip2-ipv4/master/data/geoip2-ipv4.csv"];
+  let UsersLocation = SalesforceServiceCloud
+  | where TimeGenerated > timeframe
+  | where EventType == 'Login' and LoginStatus=='LOGIN_NO_ERROR'
+  | project TimeGenerated, ClientIp, UserId, User, UserType ;
+  UsersLocation
+  | extend Dummy=1
+  | summarize count() by Hour=bin(TimeGenerated,3h), ClientIp,User, Dummy
```

Comment:
3h bin but your queryPeriod is only 20m, this does not make much sense.

Reply:
Fixed